TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -1174,24 +1181,27 @@ private void ResolvePendingPERFORMProcedures()
         {
             if (this.CurrentProgramCfgBuilder.PendingPERFORMProcedures != null)
             {
+                PerformCallRelation performCallRelation = new PerformCallRelation();
                 var clonedPerforms = new List<Tuple<PerformProcedure, SectionNode, BasicBlockForNodeGroup>>();
-
                 //First pass: resolve targets of PERFORMs, some new groups may be created during this
                 foreach (var item in this.CurrentProgramCfgBuilder.PendingPERFORMProcedures)
                 {
                     item.Item3.RecursivityGroupSet = new BitArray(GroupCounter + 1);
                     item.Item3.RecursivityGroupSet.Set(item.Item3.GroupIndex, true);
-                    ResolvePendingPERFORMProcedure(item, clonedPerforms);
+                    ResolvePendingPERFORMProcedure(item, clonedPerforms, performCallRelation);
                 }
 
                 //Second pass: resolve cloned groups created during first pass
                 for (int i = 0; i < clonedPerforms.Count; i++)
                 {
                     //We are using a regular for instead of foreach because new groups may also be created during this second pass.
                     //New groups/performs to resolve are added at tail and all are processed during this second pass
-                    ResolvePendingPERFORMProcedure(clonedPerforms[i], clonedPerforms);
+                    ResolvePendingPERFORMProcedure(clonedPerforms[i], clonedPerforms, performCallRelation);
                 }
 
+                // Report Recursive PERFORM call
+                ReportRecursivePerformCall(performCallRelation);
```

Comment:
As we have now a new mechanism to detect recursive PERFORMs, we can remove the `RecursivityGroupSet` property from `BasicBlockForNodeGroup` class. I've checked the code, it is now written only, never read.

Reply:
Done in 35d0734